src/lib/attach-shared-listeners.ts

```diff
@@ -23,6 +31,13 @@ const bindMessageHandler = (
         return;
       }
 
+      if (
+        heartbeatOptions &&
+        typeof heartbeatOptions !== "boolean" &&
+        heartbeatOptions?.kind === message.data
```

Comment:
As written in the single attach listener, this will not work in a ping/pong strategy.